driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -271,6 +302,42 @@ pub mod tests {
         assert_eq!(solution, expected_solution);
     }
 
+    #[test]
+    fn test_failed_deserialize_result() {
+        let json = json!({
+            "The Prices": {
+                "tokenA": 1,
+                "tokenB": "2",
+            },
+        });
+        let err = deserialize_result(&json).expect_err("Should fail to parse");
+        assert_eq!(err.description(), "No 'price' object in json");
+
+        let json = json!({
+            "prices": {
+                "token1": 1,
+            },
+        });
+        let err = deserialize_result(&json).expect_err("Should fail to parse");
+        assert_eq!(err.description(), "Price value not a string");
+
+        let json = json!({
+            "prices": {
+                "tokenX": "1",
+            },
+        });
+        let err = deserialize_result(&json).expect_err("Should fail to parse");
+        assert_eq!(err.description(), "Failed to parse token id");
```

Comment:
It would be great to include the `ParseIntError` in the `PriceFindingError` so that you know why the parsing failed. Here its because `X` is an unexpected character, the next case is because of an overflow.

Reply:
Is there some sweet trick I can do to recover the `ParseIntError`? Like a string format?

Reply:
It looks like PriceFindingError already implements `impl From<std::num::ParseIntError> for PriceFindingError` so you should be able to use `?` or `.map_err(PriceFindingError::from)`.

Reply:
Right, but then I wouldn't get the detailed string...

Reply:
I was trying to avoid using `PriceFindingError::from` since it returns `ErroKind::unknown`

Reply:
Really? https://github.com/gnosis/dex-services/blob/master/driver/src/price_finding/error.rs#L29 The code here suggests its an `ErrorKind::ParseIntError`.

Reply:
If you want a JSON error you can do:
```
.map_err(|err| PriceFindingError::new(err.to_string(), ErrorKind::JsonError))
```

Reply:
No no, I want parse Int error, but I wanted to be more specific about exactly what was trying to be parsed, so I had a string...

Reply:
Then may I suggest:
```
.map_err(|err| PriceFindingError::new(
    format!("error doing something: {}", err),
    ErrorKind::JsonError.
))
```

Reply:
I believe I have already done this...

Reply:
But both errors seem to have the same message: `assert_eq!(err.description(), "Failed to parse token id");`